src/main/java/com/esri/samples/na/routing_around_barriers/RoutingAroundBarriersController.java

```diff
@@ -0,0 +1,328 @@
+import com.esri.arcgisruntime.symbology.SimpleFillSymbol;
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol;
+import com.esri.arcgisruntime.symbology.TextSymbol;
+import com.esri.arcgisruntime.tasks.networkanalysis.DirectionManeuver;
+import com.esri.arcgisruntime.tasks.networkanalysis.PolygonBarrier;
+import com.esri.arcgisruntime.tasks.networkanalysis.Route;
+import com.esri.arcgisruntime.tasks.networkanalysis.RouteParameters;
+import com.esri.arcgisruntime.tasks.networkanalysis.RouteResult;
+import com.esri.arcgisruntime.tasks.networkanalysis.RouteTask;
+import com.esri.arcgisruntime.tasks.networkanalysis.Stop;
+
+public class RoutingAroundBarriersController {
+
+  @FXML private MapView mapView;
+  @FXML private ToggleButton btnAddStop;
+  @FXML private ToggleButton btnAddBarrier;
+  @FXML private Button btnDetermineRoute;
+  @FXML private Button btnReset;
+  @FXML private CheckBox findBestSequenceCheckBox;
+  @FXML private CheckBox preserveFirstStopCheckBox;
+  @FXML private CheckBox preserveLastStopCheckBox;
+  @FXML private ListView<String> directionsList;
+  @FXML private TitledPane routeInformationTitledPane;
+
+  // for displaying the route to the mapview
+  private GraphicsOverlay routeGraphicsOverlay;
+  // symbol for marking routes
+  private SimpleLineSymbol routeLineSymbol;
+  // task to find route between stops
+  private RouteTask routeTask;
+  // used for solving task above
+  private RouteParameters routeParameters;
+  // for grouping stops to add to the routing task
+  private ArrayList<Stop> stopsList;
+  // for grouping barriers to add to the routing task
+  private ArrayList<PolygonBarrier> barriersList;
+
+  @FXML
+  public void initialize() {
+    // create an ArcGISMap with a streets basemap
+    ArcGISMap map = new ArcGISMap(Basemap.createStreets());
+    // set the ArcGISMap to be displayed in this view
+    mapView.setMap(map);
+    // zoom to viewpoint
+    mapView.setViewpoint(new Viewpoint(32.727, -117.1750, 40000));
+
+    // create graphics overlays for stops, barriers and route
+    GraphicsOverlay stopsGraphicsOverlay = new GraphicsOverlay();
+    GraphicsOverlay barriersGraphicsOverlay = new GraphicsOverlay();
+    routeGraphicsOverlay = new GraphicsOverlay();
+
+    // add the graphics overlays to the map view
+    mapView.getGraphicsOverlays().addAll(Arrays.asList(stopsGraphicsOverlay, barriersGraphicsOverlay, routeGraphicsOverlay));
+
+    // create a list of stops and a list of barriers
+    stopsList = new ArrayList<>();
+    barriersList = new ArrayList<>();
+
+    // initialize the TitlePane of the Accordion box
+    routeInformationTitledPane.setText("No route to display");
+
+    // create symbols for displaying the barriers and the route line
+    SimpleFillSymbol barrierSymbol = new SimpleFillSymbol(SimpleFillSymbol.Style.DIAGONAL_CROSS, 0xFFFF0000, null);
+    routeLineSymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, 0x800000FF, 5.0f);
+
+    // disable the UI until ready
```

Comment:
these can also be moved to the FXML file, within each button's tag `disable = "true"`